### PR TITLE
Apple news retry

### PR DESCRIPTION
**Pull request by tbone1020**

### CODE REVIEW

File: src/Notify/Notifier/AppleNewsNotifier.php [hunk 1]
```diff
@@ -156,12 +156,42 @@ protected function updateArticle(Message $notification, Message $app, Message $a
...
+        if (null !== $latestRevision && $metadata['revision'] !== $latestRevision) {
+            $metadata['revision'] = $latestRevision;
+            $result = $this->api->updateArticle((string)$article->get('apple_news_id'), $document, $metadata);
+        }
+
+        if ($result['ok']) {
+            return $result;
+        }
+
+        $getArticleResult = $this->api->getArticle((string)$article->get('apple_news_id'));
```
Comment (gdbrown) [inline src/Notify/Notifier/AppleNewsNotifier.php]: @tbone1020 use fget instead of adding string casting.

File: src/Notify/Notifier/AppleNewsNotifier.php [hunk 2]
```diff
@@ -156,12 +156,42 @@ protected function updateArticle(Message $notification, Message $app, Message $a
...
+        if (!$getArticleResult['ok']) {
+            $this->sendMessageToSlack($article, $notification);
```
Comment (gdbrown) [inline src/Notify/Notifier/AppleNewsNotifier.php]: @tbone1020 instead of adding specific "sendMessageToSlack" feature, simply return result as normal and add your own `tmz:notify:node:apple-news-notification.notification-failed` event listener to your SlackWatcher and handle there.  sending to slack is a site level handling

File: src/Notify/Notifier/AppleNewsNotifier.php [hunk 3]
```diff
@@ -156,12 +156,42 @@ protected function updateArticle(Message $notification, Message $app, Message $a
...
             return $result;
         }
 
-        $metadata['revision'] = $latestRevision;
-        return $this->api->updateArticle((string)$article->get('apple_news_id'), $document, $metadata);
+        $revision = $getArticleResult['response']['data']['revision'];
+        if ($metadata['revision'] !== $revision) {
+            $metadata['revision'] = $revision;
+            $result = $this->api->updateArticle((string)$article->get('apple_news_id'), $document, $metadata);
```
Comment (gdbrown) [inline src/Notify/Notifier/AppleNewsNotifier.php]: @tbone1020 use `fget('apple_news_id')` and remove`(string)`

**Review by gdbrown — CHANGES_REQUESTED** [review]
@tbone1020 @efjacobson i'd be curious to know why apple news revision is stale.  why do we need to do this?

**Comment by efjacobson** [comment]
> @tbone1020 @efjacobson i'd be curious to know why apple news revision is stale. why do we need to do this?

@gdbrown two possibilities occur to me:
1. someone manually updated the apple news article (unlikely, perhaps impossible)
2. there is an event race condition. consider this actual sequence of events:

08:53:35 AM - apple news article is created with revision `AAAAAAAAAAD//////////w==`
08:54:01 AM - article updated
08:56:18 AM - article updated
08:59:38 AM - article updated
09:00:27 AM - apple news article synced with revision `AAAAAAAAAAAAAAAAAAAAAA==`
09:00:29 AM - apple news article synced with (same) revision `AAAAAAAAAAAAAAAAAAAAAA==`
09:22:14 AM - article updated
09:25:41 AM - article updated
09:26:28 AM - apple news article **_failed_** with revision `AAAAAAAAAAAAAAAAAAAAAQ==`
09:37:41 AM - article updated
09:43:41 AM - apple news article synced with revision `AAAAAAAAAAAAAAAAAAAAAg==`

note that the events at `09:00:27 AM` and `09:00:29 AM` are _not_ the same event that is somehow being added to the stream twice. they have different notification-sent event causators. seems to me that either
- multiples of the same notification are being scheduled (ie cancelJobs is not being called 100% of the time it should be)
- there is a race condition edge case where an immediate and a scheduled notification happen very close in time to eachother

in either case, perhaps apple returns the same revision token after each update call? something like
- update call is received, apple returns success with new revision token
- update call is received, apple returns success with same revision toke but internally queues the update which later assigns a new token to the article
- update call is received using token from first update which is now wrong

tldr i dunno, apple gon' apple

also note that in this particular case the article was _not_ fixed manually. it resolved itself, either by the existing retry mechanism or (more likely) vodou

**Comment by gdbrown** [comment]
@efjacobson i recall apple news api being buggy as hell and inconsistent.  i guess this is the least putrid option.  why do the even use a revision token if they're gonna be weird about it.

File: src/Notify/Notifier/AppleNewsNotifier.php (hunk 1 above)
Comment (tbone1020) [reply src/Notify/Notifier/AppleNewsNotifier.php]: done

File: src/Notify/Notifier/AppleNewsNotifier.php (hunk 3 above)
Comment (tbone1020) [reply src/Notify/Notifier/AppleNewsNotifier.php]: done

File: src/Notify/Notifier/AppleNewsNotifier.php (hunk 2 above)
Comment (tbone1020) [reply src/Notify/Notifier/AppleNewsNotifier.php]: @gdbrown Slack method was removed from triniti and moved it to site level.

**Comment by tbone1020** [comment]
@gdbrown ready for another review

File: CHANGELOG-1.x.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 This changelog references the relevant changes done in 1.x versions.
 
 
+## vN.N.N
```
Comment (gdbrown) [inline CHANGELOG-1.x.md]: @tbone1020 go ahead and set version to 1.1.4

Comment (gdbrown) [reply CHANGELOG-1.x.md]: @tbone1020 nvm, i'll do it.  it's quick

**Review by gdbrown — APPROVED** [review]
